### PR TITLE
[Core][Duplication] Adding deprecated message on line_2d.h

### PR DESCRIPTION
**Pull request by loumalouomega**
Related #2070 

I added message to the KRATOS_DEPRECATED macro. And used the C++14 standard if possible

### CODE REVIEW

**Comment by loumalouomega** [comment]
Looks like the message doesn't work without C++14

**Comment by loumalouomega** [comment]
Ok, that solved the problem, @jcotela , @roigcarlo ?

File: kratos/includes/define.h [hunk 1]
```diff
@@ -308,14 +308,19 @@ catch(...) { Block KRATOS_THROW_ERROR(std::runtime_error, "Unknown error", MoreI
 
     
     
-    
-#ifdef __GNUC__
+#if __cplusplus >= 201402L
```
Comment (roigcarlo) [inline kratos/includes/define.h:311]: As I told you, this is gonna fail in MSVC, but let me try it first. What were you compiling to trigger the error?

Comment (loumalouomega) [reply kratos/includes/define.h:311]: You have a elif for MSCV two lines late, and it compiled the appveyor

**Review by roigcarlo — APPROVED** [review]
:+1: But we have to think when we will deleted all the deprecateds in Kratos!

**Comment by loumalouomega** [comment]
@KratosMultiphysics/technical-committee We can think about time to remove deprecated, like 6 months

**Comment by loumalouomega** [comment]
Which remember me this 

https://github.com/KratosMultiphysics/Kratos/blob/ee695b070b276c50a18cc4047cfeb9786be09bd8/kratos/utilities/math_utils.h#L851

https://github.com/KratosMultiphysics/Kratos/blob/ee695b070b276c50a18cc4047cfeb9786be09bd8/kratos/utilities/math_utils.h#L870

We should remove it

**Comment by loumalouomega** [comment]
Done in #2090

**Pull request merged**